pkg/preemptiontoleration/preemption_toleration.go

```diff
@@ -0,0 +1,429 @@
+	if !dp.PodEligibleToPreemptOthers(preemptor, nodeLister, m[preemptor.Status.NominatedNodeName]) {
+		klog.V(5).InfoS("Pod is not eligible for more preemption", "pod", klog.KObj(preemptor))
+		return "", nil
+	}
+
+	// 2) Find all preemption candidates.
+	candidates, nodeToStatusMap, status := pl.FindCandidates(ctx, state, preemptor, m)
+	if !status.IsSuccess() {
+		return "", status
+	}
+
+	// Return a FitError only when there are no candidates that fit the preemptor.
+	if len(candidates) == 0 {
+		fitError := &framework.FitError{
+			Pod:         preemptor,
+			NumAllNodes: len(nodeToStatusMap),
+			Diagnosis: framework.Diagnosis{
+				NodeToStatusMap: nodeToStatusMap,
+				// Leave FailedPlugins as nil as it won't be used on moving Pods.
+			},
+		}
+		return "", framework.NewStatus(framework.Unschedulable, fitError.Error())
+	}
+
+	// 3) Interact with registered Extenders to filter out some candidates if needed.
+	candidates, status = dp.CallExtenders(pl.fh.Extenders(), preemptor, nodeLister, candidates)
+	if !status.IsSuccess() {
+		return "", status
+	}
+
+	// 4) Find the best candidate.
+	bestCandidate := dp.SelectCandidate(candidates)
+	if bestCandidate == nil || len(bestCandidate.Name()) == 0 {
+		return "", nil
+	}
+
+	// 5) Perform preparation work before nominating the selected candidate.
+	if status := dp.PrepareCandidate(bestCandidate, pl.fh, cs, preemptor, pl.Name()); !status.IsSuccess() {
+		return "", status
+	}
+
+	return bestCandidate.Name(), nil
+}
+
+// ToleratedFromPreemption evaluates whether the victimCandidate
+// pod can tolerate from preemption by the preemptor pod or not
+// by inspecting PriorityClass of victimCandidate pod.
+// The function is public because other plugin can evaluate preemption toleration policy
+// This would be useful other PostFilter plugin depends on the preemption toleration feature.
+func ToleratedFromPreemption(
+	victimCandidate, preemptor *v1.Pod,
+	pcLister schedulinglisters.PriorityClassLister,
+	now time.Time,
+) (bool, error) {
+	if victimCandidate.Spec.PriorityClassName == "" {
+		return false, nil
+	}
+	victimPriorityClass, err := pcLister.Get(victimCandidate.Spec.PriorityClassName)
+	if err != nil {
+		return false, err
+	}
+
+	preemptorPreemptionPolicy := v1.PreemptLowerPriority
+	if preemptor.Spec.PreemptionPolicy != nil {
+		preemptorPreemptionPolicy = *preemptor.Spec.PreemptionPolicy
+	}
+	if preemptorPreemptionPolicy == v1.PreemptNever {
+		return true, nil
+	}
+
+	// check it can tolerate the preemption in terms of priority value
+	policy, err := getCompletedPreemptionToleration(*victimPriorityClass)
+	if err != nil {
+		// if any error raised, no toleration at all
+		klog.ErrorS(err, "Failed to parse preemption toleration policy of victim candidate's priorityclass.  This victim candidate can't tolerate the preemption",
+			"PreemptorPod", klog.KRef(preemptor.Namespace, preemptor.Name),
+			"VictimCandidatePod", klog.KRef(victimCandidate.Namespace, victimCandidate.Name),
```

Comment:
```suggestion
			"PreemptorPod", klog.KObj(preemptor),
			"VictimCandidatePod", klog.KObj(victimCandidate),
```

Reply:
fixed in https://github.com/kubernetes-sigs/scheduler-plugins/pull/224/commits/18a02054d1fe96234351f4dc5e22d7430fd140dd